esphome/components/ld2450/ld2450.cpp

```diff
@@ -0,0 +1,827 @@
+        index.is_moving == is_moving) {
+      count++;
+    }
+  }
+  return count;
+}
+
+// Service reset_radar_zone
+void LD2450Component::reset_radar_zone() {
+  this->zone_type_ = 0;
+  for (auto &i : zone_config_) {
+    i.x1 = 0;
+    i.y1 = 0;
+    i.x2 = 0;
+    i.y2 = 0;
+  }
+  this->send_set_zone_command_();
+}
+
+void LD2450Component::set_radar_zone(int32_t zone_type, int32_t zone1_x1, int32_t zone1_y1, int32_t zone1_x2,
+                                     int32_t zone1_y2, int32_t zone2_x1, int32_t zone2_y1, int32_t zone2_x2,
+                                     int32_t zone2_y2, int32_t zone3_x1, int32_t zone3_y1, int32_t zone3_x2,
+                                     int32_t zone3_y2) {
+  this->zone_type_ = zone_type;
+  int zone_parameters[12] = {zone1_x1, zone1_y1, zone1_x2, zone1_y2, zone2_x1, zone2_y1,
+                             zone2_x2, zone2_y2, zone3_x1, zone3_y1, zone3_x2, zone3_y2};
+  for (int i = 0; i < MAX_ZONES; i++) {
+    zone_config_[i].x1 = zone_parameters[i * 4];
+    zone_config_[i].y1 = zone_parameters[i * 4 + 1];
+    zone_config_[i].x2 = zone_parameters[i * 4 + 2];
+    zone_config_[i].y2 = zone_parameters[i * 4 + 3];
+  }
+  this->send_set_zone_command_();
+}
+
+// Set Zone on LD2450 Sensor
+void LD2450Component::send_set_zone_command_() {
+  uint8_t cmd_value[26] = {};
+  uint8_t zone_type_bytes[2] = {static_cast<uint8_t>(this->zone_type_), 0x00};
+  uint8_t area_config[24] = {};
+  for (int i = 0; i < MAX_ZONES; i++) {
+    int values[4] = {zone_config_[i].x1, zone_config_[i].y1, zone_config_[i].x2, zone_config_[i].y2};
+    this->convert_int_values_to_hex_(values, area_config + (i * 8));
+  }
+  std::memcpy(cmd_value, zone_type_bytes, 2);
+  std::memcpy(cmd_value + 2, area_config, 24);
+  set_config_mode_(true);
+  send_command_(CMD_SET_ZONE, cmd_value, 26);
+  set_config_mode_(false);
+}
+
+// Convert signed int to HEX high and low bytes
+void LD2450Component::convert_int_values_to_hex_(const int *values, uint8_t *bytes) {
+  for (int i = 0; i < 4; i++) {
+    std::string temp_hex = convert_signed_int_to_hex_(values[i]);
+    bytes[i * 2] = std::stoi(temp_hex.substr(2, 2), nullptr, 16);      // Store high byte
+    bytes[i * 2 + 1] = std::stoi(temp_hex.substr(0, 2), nullptr, 16);  // Store low byte
+  }
+}
+
+// Check presense timeout to reset presence status
+bool LD2450Component::get_timeout_status_(int32_t check_millis) {
+  if (check_millis == 0)
+    return true;
+  if (this->timeout_ == 0)
+    this->timeout_ = this->convert_seconds_to_ms(DEFAULT_PRESENCE_TIMEOUT);
+  int32_t current_millis = millis();
+  int32_t timeout = this->timeout_;
+  return current_millis - check_millis >= timeout;
+}
+
+// Extract, store and publish zone details LD2450 buffer
+void LD2450Component::process_zone_(uint8_t *buffer) {
+  uint8_t index, start;
+  for (index = 0; index < MAX_ZONES; index++) {
+    start = 12 + index * 8;
+    zone_config_[index].x1 = this->hex_to_signed_int_(buffer, start);
```

Comment:
> All uses of class members and member functions should be prefixed with `this->` to distinguish them from global functions/variables.

https://esphome.io/guides/contributing#codebase-standards
```suggestion
    this->zone_config_[index].x1 = this->hex_to_signed_int_(buffer, start);
```